packages/truffle/test/scenarios/commandRunner.js

```diff
@@ -74,17 +79,20 @@ module.exports = {
 
       child.stdout.on("data", data => {
         // accumulate buffer from chunks
-        if (!seenChildPrompt) {
-          outputBuffer += data;
-        }
+        outputBuffer += data;
 
-        // child process is ready for input when it displays the readyPrompt
-        if (!seenChildPrompt && outputBuffer.includes(readyPrompt)) {
-          seenChildPrompt = true;
-          inputCommands.forEach(command => {
-            child.stdin.write(command + EOL);
-          });
-          child.stdin.end();
+        // count prompt
+        const foundPrompts = (outputBuffer.match(readyPromptRex) || []).length;
+        if (foundPrompts > numSeenPrompts) {
+          numSeenPrompts = foundPrompts;
```

Comment:
Might this be better? It
- Doesn't tire as `outputBuffer` grows
- Removes equivocation between `found` and `numSeen`

```suggestion
        const numNewPrompts = (data.match(readyPromptRex) || []).length;
        if (numNewPrompts > 0) {
          numSeenPrompts = += numNewPrompts;
```

Reply:
@cliffoo `data` could be a _partial_ line, so this method could potentially miss lines completely.

In other words: the `data` event could be emitted multiple times for a line like "truffle(develop)>\n":

1. `truffle<de`
2. `velop>\n`